internal/pkg/cacheutil/processcache.go

```diff
@@ -32,27 +33,32 @@ func (shard *ShardedIncomingDataCache) FlushPrometheusMetric(usetimestamp bool,
 	defer shard.lock.Unlock()
 	minMetricCreated := 0 //..minimum of one metrics created
 
-	for _, IncomingDataInterface := range shard.plugin {
-		if collectd, ok := IncomingDataInterface.(*incoming.CollectdMetric); ok {
-			if collectd.ISNew() {
-				collectd.SetNew(false)
-				for index := range collectd.Values {
-					m, err := tsdb.NewCollectdMetric(usetimestamp, *collectd, index)
-					if err != nil {
-						log.Printf("newMetric: %v", err)
-						continue
-					}
-					ch <- m
-					minMetricCreated++
-				}
-			} else {
-				//clean up if data is not access for max TTL specified
-				if shard.Expired() {
-					delete(shard.plugin, collectd.GetItemKey())
-					//log.Printf("Cleaned up plugin for %s", collectd.GetKey())
+	for _, dataInterface := range shard.plugin {
+		format := saconfig.DataSourceUniversal.String()
+		switch metric := dataInterface.(type) {
```

Comment:
Why not just add the String() method to the MetricDataFormat interface so that you don't have to type switch just to get the formatted string? This seems like something this type should always implement.

Reply:
Good point, will do.